pkg/controller/infinispan/infinispan_controller.go

```diff
@@ -1305,28 +1318,42 @@ func (r *ReconcileInfinispan) reconcileGracefulShutdown(ispn *infinispanv1.Infin
 func (r *ReconcileInfinispan) gracefulShutdownReq(ispn *infinispanv1.Infinispan, podList *corev1.PodList,
 	logger logr.Logger, cluster ispn.ClusterInterface) (*reconcile.Result, error) {
 	logger.Info("Sending graceful shutdown request")
-	// send a graceful shutdown to the first ready pod
-	// if there's no ready pod we're in trouble
+	// Send a graceful shutdown to the first ready pod, if there's no ready pod then terminate any pending pods
+	forceShutdown := true
 	for _, pod := range podList.Items {
 		if kube.IsPodReady(pod) {
-			err := cluster.GracefulShutdown(pod.GetName())
-			if err != nil {
-				logger.Error(err, "failed to exec shutdown command on pod")
-				continue
+			if err := cluster.GracefulShutdown(pod.GetName()); err != nil {
+				logger.Error(err, "Error encountered on cluster shutdown")
+
+				// ISPN-13141 causes GracefulShutdown to fail if there are issues with the server
+				logger.Info("Cluster Shutdown failed. Attempting to execute GracefulShutdownTask")
```

Comment:
@ryanemerson I'm not sure, but probably we need to execute shutdown via task in specific case only, not for every error. As error could say about communication problem, not server.

Reply:
Ideally yes, but the problem is identifying when we should/shouldn't execute the task. If the error is caused by a communication problem, then `GracefulShutdownTask` will also fail, so we're only executing a single http request unnecessarily.

IMO I think it's better to have the additional HTTP request instead of risking getting the conditional check wrong and not executing  `GracefulShutdownTask` when it's required.